### ISSUE TITLE
Plugin resolution ignores exclusion of transitive dependencies in POM

### ISSUE THREAD

**Issue by stoyicker** [gradle/gradle#5092]
### Expected Behavior
When I import a plugin that has a POM such as this one: https://plugins.gradle.org/m2/org/stoyicker/android-check/plugin/2.222/plugin-2.222.pom
I expect Gradle to interpret this statement:
```
classpath "org.stoyicker.android-check:plugin:2.221"
``` 
as, implicitly, this:
```
classpath("org.stoyicker.android-check:plugin:2.221") {
    exclude module: "asm"
    exclude module: "gson"
    exclude module: "guava"
    exclude module: "commons-logging"
}
```

### Current Behavior
```
classpath "org.stoyicker.android-check:plugin:2.221"
```
will bring all of its transitive dependencies recursively, completely ignoring the POM definition. However, being explicit with the exclusions in Gradle, that is, importing the plugin as:
```
classpath("org.stoyicker.android-check:plugin:2.221") {
    exclude module: "asm"
    exclude module: "gson"
    exclude module: "guava"
    exclude module: "commons-logging"
}
```
will work just as expected, which makes me believe that there is some issue with POM parsing, rather than with the capabilities of excluding dependencies.

### Context
I am trying to use two plugins in a project. One is the Gradle Plugin for Android, which I'm not in control of and which has serious restrictions regarding the versions of its dependencies, meaning that it will fail to work correctly if someone tries to change the versions of any of the dependencies it relies on to a project it is applied to via something like configurations.all for example.
The other one is a plugin I develop which can be found at https://github.com/stoyicker/android-check-2
Transitively, this plugin has some common dependencies with the Gradle Plugin for Android, but on different versions. For this reason, and because for some reason I can't really understand anyone thought would ever be a good idea Gradle seems to share loaded classes between all plugins in the project, [in the Gradle configuration of the plugin you can find how I exclude some of these dependencies and instead force the plugin to compile against versions matching those of the Gradle Plugin for Android](https://github.com/stoyicker/android-check-2/blob/c2e4e5da8d501d1c21ecca9423b035324cc7bd50/plugin/build.gradle), which I trust to bring the symbols once both of them are applied to a project, since it's not possible to have two versions of the same artifact coexist. Because these dependencies are defined as exclusions in the POM of my plugin in the Gradle plugin portal, which I linked at the beginning of this issue, I assume it is reasonable to expect both plugins to work just fine (under the assumption that my plugin is compatible with the version replacements enforced by the Gradle Plugin for Android, of course, which it is) without having to repeat the exclusions explicitly in the project the plugins are being applied to.

### Steps to Reproduce
https://github.com/stoyicker/android-check-2/
This project contains two subprojects:
* plugin, which is the second one of the plugins I described above.
* sample, which I use to test the plugin. Normally I compile the plugin from `mavenLocal()`. 

Feel free to clone this project, change the `mavenLocal()` repository to `maven { url "https://plugins.gradle.org/m2/" }` and then try to run the lint task in the sample project. If you leave everything as it is, you will receive a lot of "Broken class" warnings from the Android Gradle Plugin and/or a warning about lint being forced to use a different Guava version that it should be using. However, if you explicitly add the exclusions to the classpath statement, as shown earlier in this issue, lint will work just fine.

### Your Environment
Ubuntu 16.04
Gradle wrapper 4.6
OpenJDK 1.8.0_162
gradle props: org.gradle.jvmargs=-Xmx1536M

**Comment by ljacomet** [comment]
Looks like there is an issue with parsing POM files where the exclusion is defined with only the `groupId` or `artifactId`.
As this is a valid format and even one that Gradle publishes when exclusions are defined in Gradle with only the `group` or `module`, this should be fixed.
The good news is that according to quick tests, the change required for the fix looks like a small one.

Will update once I have confirmation it does not break something else.

**Comment by ljacomet** [comment]
This is now fixed and will be part of Gradle 4.8. Note that it may change resolution results for existing builds, as such it will be listed in the release notes as a potentially breaking change.
